backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/KonanLoweringPhases.kt

```diff
@@ -86,20 +89,57 @@ internal val arrayConstructorPhase = makeKonanModuleLoweringPhase(
         description = "Transform `Array(size) { index -> value }` into a loop"
 )
 
+internal val lateinitPhase = makeKonanModuleOpPhase(
+        { context, irModule ->
+            NullableFieldsForLateinitCreationLowering(context).lower(irModule)
+            NullableFieldsDeclarationLowering(context).lower(irModule)
+            LateinitUsageLowering(context).lower(irModule)
+        },
+        name = "Lateinit",
+        description = "Lateinit properties lowering"
+)
+
+internal val sharedVariablesPhase = makeKonanModuleLoweringPhase(
+        ::SharedVariablesLowering,
+        name = "SharedVariables",
+        description = "Shared variable lowering",
+        prerequisite = setOf(lateinitPhase)
```

Comment:
Do I understand correctly that this dependency exists because of local `lateinit` variables? If shared variables are lowered first, then `LateinitUsageLowering` can't properly detect reads from "shared" `lateinit` variables, right?

Reply:
Yeah, I have also considered an alternative - save `lateinit` modifier for shared variables, but it's worse, because we need to initialize them somewhere or replace simple code like `shared.element = value` with `shared = Ref(value)`.